sled-agent/src/nexus.rs

```diff
@@ -39,27 +39,57 @@ pub struct LazyNexusClient {
 }
 
 impl LazyNexusClient {
-    pub fn new(log: Logger, addr: Ipv6Addr) -> Result<Self, ResolveError> {
+    pub fn new_from_subnet(
+        log: Logger,
+        addr: Ipv6Addr,
+    ) -> Result<Self, ResolveError> {
         Ok(Self {
             inner: Arc::new(Inner {
                 log: log.clone(),
                 resolver: Resolver::new_from_ip(
                     log.new(o!("component" => "DnsResolver")),
                     addr,
                 )?,
+                address: Mutex::new(None),
+            }),
+        })
+    }
+
+    pub fn new_from_dns(
+        log: Logger,
+        dns_addrs: Vec<SocketAddr>,
+    ) -> Result<Self, ResolveError> {
```

Comment:
I wonder if it would better separate the concerns to have the caller provide a `Resolver` directly here.

Reply:
Done (indirectly), see other comments about how I'm passing the resolver.